packages/internet-header/src/services/search/coveo.service.ts

```diff
@@ -15,25 +17,36 @@ export const getCoveoSuggestions = async (query: string): Promise<CoveoCompletio
   const config = state.localizedConfig.header.search;
   const { token, organisation } = coveo.environment[state.environment];
   const url = `${coveo.url}?q=${query}&locale=${state.currentLanguage}&searchHub=${config.searchHubName}&pipeline=${config.searchPipelineName}&organizationId=${organisation}`;
-  let coveoCompletions: CoveoCompletion[] | undefined;
 
-  try {
-    const coveoResponse = await fetch(url, {
+  if (suggestionsController) suggestionsController.abort();
+  suggestionsController = new AbortController();
+
+  return new Promise((resolve, reject) => {
+    fetch(url, {
+      signal: suggestionsController.signal,
       method: 'GET',
       headers: {
         Authorization: `Bearer ${token}`,
       },
-    });
-    const coveoJSON = (await coveoResponse.json()) as CoveoResponse;
-    coveoCompletions = coveoJSON.completions.map(completion => ({
-      ...completion,
-      redirectUrl: getSearchPageUrl(completion.expression),
-    }));
-  } catch (error) {
-    console.error(
-      'Connection to coveo failed. Did you add "*.coveo.com" to your connect-src content security policy and tried turning off your adblocker?',
-    );
-  }
-
-  return coveoCompletions === undefined ? [] : coveoCompletions;
+    })
+      .then(response => response.json())
+      .then((coveoJSON: CoveoResponse) => {
+        resolve(
+          coveoJSON.completions.map(completion => ({
+            ...completion,
+            redirectUrl: getSearchPageUrl(completion.expression),
+          })),
+        );
+      })
+      .catch(error => {
+        if (error.name === 'AbortError') {
+          console.warn('Coveo suggestions request aborted.');
+        } else {
```

Comment:
I'd not log abort errors as they are expected to happen. This will just spam the console.

```suggestion
        if (error.name !== 'AbortError') {
```